java/ql/src/semmle/code/java/Statement.qll

```diff
@@ -201,10 +201,10 @@ class ForStmt extends ConditionalStmt, @forstmt {
     getCondition().getAChildExpr*() = result.getAnAccess()
   }
 
-  /** Gets a printable representation of this statement. May include more detail than `toString()`. */
   override string pp() { result = "for (...;...;...) " + this.getStmt().pp() }
```

Comment:
Guessing the reason for difference between `for (...;...;...)` and `for (...)` is for-each loops? If so we should probably fix the pp() output as well.

Reply:
Do you mean this `pp()` result should be changed? Maybe it would be more correct to change `toString()` to be `for (...;...;...)`  as well, since it looks like a `ForStmt` is never an `EnhancedForStmt`.

Reply:
If it definitely is of the form `for(...;...;...)` then go for that

Reply:
At least the Query Console says that `ForStmt` and `EnhancedForStmt` are incompatible.
I have changed `toString()` now in the latest commit to use `for(...;...;...)`.